PostHog/Utils/UIApplication+.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  UIApplication+.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 11/11/2024.
+//
+
+#if os(iOS) || os(tvOS)
+    import UIKit
+
+    extension UIApplication {
+        static func getCurrentWindow(filterForegrounded: Bool = true) -> UIWindow? {
+            let windowScenes = UIApplication.shared
+                .connectedScenes
+                .compactMap { $0 as? UIWindowScene }
+                .filter {
+                    !filterForegrounded || $0.activationState == .foregroundActive
+                }
+
+            for scene in windowScenes {
+                // attempt to retrieve directly from UIWindowScene
+                if #available(iOS 15.0, tvOS 15.0, *) {
+                    if let keyWindow = scene.keyWindow {
+                        return keyWindow
+                    }
+                }
+
+                // check scene.windows.isKeyWindow
+                for window in scene.windows {
+                    if window.isKeyWindow {
+                        return window
+                    }
+                }
```

Comment:
can this be an else branch for `if #available`? otherwise its being executed twice for iOS >= 15, the former and this one when its not a keyWindow.